### PR TITLE
pushbullet: Email address and multiple API key support

### PR DESCRIPTION
**Pull request by nikdoof**
Added the ability to specify email address destinations (as dictated by Pushbullet's API) and also support for sending messages via multiple API keys.

Also, I padded out the requests with more request information to ensure the queries are interpreted correctly and the correct response is received.

### CODE REVIEW

File: flexget/plugins/output/pushbullet.py [hunk 1]
```diff
@@ -45,31 +47,26 @@ class OutputPushbullet(object):
...
-            log.info('    Type: %s' % push_type)
-            log.info('    Device: %s' % devices)
-
-        # Loop through the provided entries
         for entry in task.accepted:
-
             title = config['title']
             body = config['body']
-            url = config.get('url', '')
+            url = config.get('url', None)
```
Comment (gazpachoking) [inline flexget/plugins/output/pushbullet.py]: `None` is the default default for get method, no need to specify explicitly.

File: flexget/plugins/output/pushbullet.py [hunk 2]
```diff
@@ -86,48 +83,74 @@ def on_task_output(self, task, config):
...
-                if request_status == 200:
-                    log.debug('Pushbullet notification sent')
-                elif request_status == 500:
-                    log.warning('Pushbullet notification failed, Pushbullet API having issues')
-                    #TODO: Implement retrying. API requests 5 seconds between retries.
-                elif request_status >= 400:
-                    error = json.loads(response.content)['error']
-                    log.error('Pushbullet API error: %s' % error['message'])
+            for apikey in apikeys:
+                if len(destinations):
```
Comment (gazpachoking) [inline flexget/plugins/output/pushbullet.py]: No need for `len` here, a list is considered false if it is empty.

File: flexget/plugins/output/pushbullet.py [hunk 3]
```diff
@@ -45,31 +47,26 @@ class OutputPushbullet(object):
...
-        push_type = 'link' if config.has_key('url') else 'note'
+        emails = config.get('email', [])
+        if not isinstance(emails, list):
+            emails = [emails]
 
-        # Set a bunch of local variables from the config
-        apikey = config['apikey']
+        destinations = []
+        for typ, val in [('device_iden', devices), ('email', emails)]:
+            destinations.extend([(x, typ) for x in val])
```
Comment (gazpachoking) [inline flexget/plugins/output/pushbullet.py]: This is a bit confusing, I think it might be a bit more clear just making a separate loop for each type.

**Comment by nikdoof** [comment]
Sorry for letting this pull rest for so long, i've got quite a few improvements i've made since, for the moment i'm going to close this pull and resubmit once i've completed my modifications and cleanups.

**Pull request closed without merging**